src/lib.rs

```diff
@@ -89,6 +94,30 @@ fn read<T: io::Read>(port: &mut T, len: i32) -> Result<(Vec<u8>)> {
     return Ok(vec);
 }
 
+fn read_once<T: io::Read>(port: &mut T, len: i32) -> Result<(Vec<u8>)> {
+    let mut vec: Vec<u8> = vec![];
+    let mut len = len;
+
+    loop {
+        let buf: &mut [u8; 1] = &mut [0u8];
+
+        match port.read(buf) {
+            Ok(_) => {
+                vec.push(buf[0]);
+                len = len - 1;
+                if len == 0 {
+                    break;
+                }
+            }
+            Err(e) => {
```

Comment:
I think this error block could be written like this:

```suggestion
            Err(e) => Err(Error::new(ErrorKind::Other, ""))
```

Reply:
Done!